### PR TITLE
Move creation of selected git commands from `GitModule`

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

- Move tests to match folder structure
- Sort implementations and tests lexicographically
- Move GitModule.Push* commands
- Move GitModule.Commit commands

## Merge strategy

❗  DO NOT SQUASH

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by gerhardol** [comment]
Some tests like those in UnitTests/GitCommands.Tests/Git/CommandsTests.cs seem to be removed. Is that intentional?

**Comment by RussKie** [comment]
I didn't delete any tests, just moved those around.
But I'll double check

**Review by mstv — COMMENTED** [review]
It just looks like lots of tests had been removed in the first commit. That is because testcases have been moved out to a new file:

```
--- a/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
+++ b/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs

--- a/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
+++ b/UnitTests/GitCommands.Tests/Git/CommandsTests.cs
```

image

---

But what is the criterion for this split of `GitCommandHelpersTest.cs`?

---

> Move tests to match folder structure

  For technical reasons (avoiding ambiguity), the namespace name was tinkered. 

`namespace GitCommandsTests.Git_Commands`
means
`namespace GitCommandsTests.Git.Commands`

(If you think this is a problem, we should rather switch to `namespace UnitTests.GitCommands_.Git_.Commands_` for instance.)

Hence the "git tag" tests should not be moved up out of the folder `UnitTests/GitCommands.Tests/Git/Commands/`:

```
--- a/UnitTests/GitCommands.Tests/Git/Commands/GitCreateTagCmdTests.cs
+++ b/UnitTests/GitCommands.Tests/Git/CommandsTests.CreateTag.cs
@@ -4,8 +4,7 @@
 
 namespace GitCommandsTests.Git_Commands
 {
-    [TestFixture]
-    public sealed class GitCreateTagCmdTests
+    partial class CommandsTests
```

---

🤔 Test names have lost the relation to the command:

Before | After
-|-
image | image

**Comment by RussKie** [comment]
> But what is the criterion for this split of `GitCommandHelpersTest.cs`?

It was easier to rename GitDeleteBranchCmdTests.cs to CommandsTests.DeleteBranch.cs than to move the content and make the review harder. The same applies to CommandsTests.CreateTag.cs.
These files contain tests for `Commands.DeleteBranch` and `Commands.CreateTag` respectively that are defined in GitCommands/Git/Commands.cs; and the naming/placement reflect the location of the shippable code.

**Comment by RussKie** [comment]
> Test names have lost the relation to the command:
> 
> Before	After
> image	image

This means the original tests were poorly named, and I didn't pay much attention to those. My goal was to consolidate the tests and place those in the correct folder structure.

**Comment by RussKie** [comment]
> It just looks like lots of tests had been removed in the first commit. That is because testcases have been moved out to a new file:
> 
> ```
> --- a/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
> +++ b/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
> 
> --- a/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
> +++ b/UnitTests/GitCommands.Tests/Git/CommandsTests.cs
> ```
> 
> image

GitCommandHelpersTest contain tests for commands defined in GitModule class. Those have to be moved out (#11308 is contributing to that. However, there are few methods like `FetchCmd` and `PullCmd` that rely on GitModule's internals, and I'm yet to figure out how to break that dependency); and once those are moved out the test will be moved out from  GitCommandHelpersTest.cs as well.

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 1]
```diff
@@ -1375,11 +1375,12 @@ void DoCommit()
                         return;
                     }
 
-                    ArgumentString commitCmd = Module.CommitCmd(
+                    ArgumentString commitCmd = Commands.Commit(
                         createAmendCommit,
                         signOffToolStripMenuItem.Checked,
                         toolAuthor.Text,
                         _useFormCommitMessage,
+                        Module.GetGitExecPath(Path.Combine(Module.WorkingDirGitDir, "COMMITMESSAGE")),
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormCommit.cs:1383]:
```suggestion
                        _commitMessageManager.CommitMessagePath,
```

**Review by mstv — COMMENTED** [review]
LGTM

File: GitCommands/Git/GitModule.cs [hunk 2]
```diff
@@ -2041,22 +1984,6 @@ public bool InTheMiddleOfAction()
...
-        {
-            return new GitArgumentBuilder("commit")
-            {
-                { amend, "--amend" },
-                { noVerify, "--no-verify" },
-                { signOff, "--signoff" },
-                { !string.IsNullOrEmpty(author), $"--author=\"{author?.Trim().Trim('"')}\"" },
-                { gpgSign && string.IsNullOrWhiteSpace(gpgKeyId), "-S" },
-                { gpgSign && !string.IsNullOrWhiteSpace(gpgKeyId), $"-S{gpgKeyId}" },
-                { useExplicitCommitMessage, $"-F \"{GetGitExecPath(Path.Combine(GetGitDirectory(), "COMMITMESSAGE"))}\"" },
```
Comment (gerhardol) [inline GitCommands/Git/GitModule.cs:2054]: This change prevents commits in WSL.
Only in master.
Not sure if other commands are affected, refactoring moving code are hard to review.

Comment (mstv) [reply GitCommands/Git/GitModule.cs:2054]: This is the only real change. A comment would have been good. How does this make it work in WSL?

Comment (gerhardol) [reply GitCommands/Git/GitModule.cs:2054]: The path is in the format for gitexec, ie in wsl format.
Added for a few commands, manually identified.

This was a series of changes, I did not see any related changes. So hopefully the only issue. 

The transformation will have to be added to the caller of commands arguments where module is available.

**Pull request merged**